File: Labs/Lab 8/Lab8.java
public class Lab8 {
    public int coinChange(int[] coins, int amount) {
        int [] dpArray = new int[amount + 1];
        for (int i = 1; i < dpArray.length; i++){
            dpArray[i] = Integer.MAX_VALUE;
        }
        for (int i = 0; i < coins.length; i++){
            for (int j = coins[i]; j < dpArray.length; j++){
                if (dpArray[j - coins[i]] != Integer.MAX_VALUE){
                    if (dpArray[j - coins[i]] < dpArray[j]){
                        dpArray[j] = dpArray[j - coins[i]] + 1;
                    }
                }
            }
        }
        if (dpArray[amount] == Integer.MAX_VALUE){
            return -1;
        }
        else{
            return dpArray[amount];
        }
    }
}

File: Labs/Lab 8/Lab8.py
class Lab8(object):
    def coinChange(self, coins: List[int], amount: int) -> int:
        dpArray = [float('inf')] * (amount + 1)
        dpArray[0] = 0
        for i in range(len(coins)):
            for j in range(coins[i], amount + 1):
                if dpArray[j - coins[i] != float('inf')]:
                    if dpArray[j - coins[i]] < dpArray[j]:
                        dpArray[j] = dpArray[j - coins[i]] + 1
        if dpArray[amount] == float('inf'):
            return -1
        else:
            return dpArray[amount]


